tests/cephfs/cephfs_mirroring/test_cephfs_mirroring_ha_metrics.py

```diff
@@ -0,0 +1,885 @@
+            initial_metrics_data, cephfs_mirror_node2
+        )
+        log.info(f"CephFS Mirror Metrics for mirror_node1 : {initial_metrics_data}")
+        cephfs_mirror_node2_data = json.loads(mirror_node2_data)
+
+        log.info(
+            "Validate if all the Details of Mirrored Filesystem is displayed in the Counters"
+        )
+        resource_name = "cephfs_mirror_mirrored_filesystems"
+        filesystem_name = source_fs
+        label_on_node1, counter_on_node1 = fs_mirroring_utils.get_labels_and_counters(
+            resource_name, filesystem_name, cephfs_mirror_node1_data
+        )
+        label_on_node2, counter_on_node2 = fs_mirroring_utils.get_labels_and_counters(
+            resource_name, filesystem_name, cephfs_mirror_node2_data
+        )
+
+        peers_on_node1 = counter_on_node1.get("mirroring_peers")
+        peers_on_node2 = counter_on_node2.get("mirroring_peers")
+        if peer_connection == peers_on_node1 and peers_on_node2:
+            log.info(
+                f"Configured Peer : {peer_connection} matches in Metrics : "
+                f"{peers_on_node1} and {peers_on_node2}"
+            )
+        else:
+            log.error(
+                f"Configured Peer : {peer_connection} doesn't exist in Metrics: "
+                f"{peers_on_node1} and {peers_on_node2}"
+            )
+            raise CommandFailed("Configured Mirroring Peers is mismatching in Metrics")
+
+        dir_count_on_node1 = counter_on_node1.get("directory_count")
+        dir_count_on_node2 = counter_on_node2.get("directory_count")
+
+        if directory_count == dir_count_on_node1 and dir_count_on_node2:
+            log.info(
+                f"Enabled directory count : {directory_count} matches in Metrics : "
+                f"{dir_count_on_node1} and {dir_count_on_node2}"
+            )
+        else:
+            log.error(
+                f"Enabled directory count : {directory_count} doesn't match in Metrics : "
+                f"{dir_count_on_node1} and {dir_count_on_node2}"
+            )
+            raise CommandFailed("Configured Directory Counts is mismatching in Metrics")
+
+        log.info(
+            "Validate if all the Details of Peer Connection is displayed in the Counters"
+        )
+        resource_name = "cephfs_mirror_peers"
+        filesystem_name = source_fs
+        label_on_node1, counter_on_node1 = fs_mirroring_utils.get_labels_and_counters(
+            resource_name, filesystem_name, cephfs_mirror_node1_data
+        )
+        label_on_node2, counter_on_node2 = fs_mirroring_utils.get_labels_and_counters(
+            resource_name, filesystem_name, cephfs_mirror_node2_data
+        )
+
+        peer_cluster_filesystem_on_node1 = label_on_node1.get("peer_cluster_filesystem")
+        peer_cluster_filesystem_on_node2 = label_on_node2.get("peer_cluster_filesystem")
+        if (
+            target_fs == peer_cluster_filesystem_on_node1
```

Comment:
We need to change the text validation here, because even if peer_cluster_filesystem_on_node1 = cephfs and peer_cluster_filesystem_on_node2=cephfs1 , the if condition will result in True, because target_fs is compared against only peer_cluster_filesystem_on_node1 and not peer_cluster_filesystem_on_node2. 
Verified it offline.

Reply:
Fixed it